file/file.h

```diff
@@ -204,7 +207,7 @@ struct FileSystem {
     }
 #else
     DIR* dir = ::opendir(directory.c_str());
-    const auto closedir_guard = MakeScopeGuard([dir]() { ::closedir(dir); });
+    const auto closedir_guard = MakeScopeGuard([dir]() { if (dir) ::closedir(dir); });
```

Comment:
Thanks Max! Could you please put `{ ::closedir(dir); }` to make me sleep better? :)